pkg/controller/cloud/node_controller.go

```diff
@@ -62,17 +62,22 @@ func NewCloudNodeController(
 	nodeInformer coreinformers.NodeInformer,
 	kubeClient clientset.Interface,
 	cloud cloudprovider.Interface,
-	nodeStatusUpdateFrequency time.Duration) *CloudNodeController {
+	nodeStatusUpdateFrequency time.Duration) (*CloudNodeController, error) {
+
+	if kubeClient == nil {
+		return nil, errors.New("kubernetes client is nil")
+	}
+
+	if cloud == nil {
+		return nil, errors.New("no cloud provider provided")
```

Comment:
Something has gone very wrong to get here. Failure to init the cloud provider in the CCM is a fatal error...

Reply:
Yes, `cloud` has been checked before passing to the constructor, and `kubeClient` is created by `ctx.ClientBuilder.ClientOrDie("node-controller")`

https://github.com/kubernetes/kubernetes/blob/3a501844219d06b6fd777a911175b823ab146d43/pkg/controller/client_builder.go#L81-L87

so, we don't need to check `cloud` and `kubeClient` in the controller initializers.

Reply:
Agree we probably don't need those checks but probably doesn't hurt to keep them

Reply:
emmm, I think it could be viewed as a cleanup for codes. There are many controllers consist of `cloud` and `kubeClient`, we don't need to check them every time.

Reply:
Fine with me :)